File: landing-react/src/lib/anbieter.ts
/**
 * Anbieterdaten nach § 5 DDG. Eine Quelle für alle Rechtsseiten.
 *
 * Bis zum 01.09.2026 trugen /impressum, /agb und /datenschutz drei getrennte
 * Sätze erfundener Angaben: "Complyo GmbH", "Musterstraße 123, 10115 Berlin",
 * "Max Mustermann", eine erfundene Handelsregisternummer, eine erfundene
 * USt-IdNr. und die Telefonnummer "+49 (0) 30 1234567". Das Impressum wurde
 * zuerst entgiftet, AGB und Datenschutzerklärung behielten die Platzhalter.
 * Drei Quellen, von denen eine gepflegt wurde, sind der Grund dafür.
 *
 * Deshalb steht alles hier. Wer die Anbieterdaten ändert, ändert sie damit für
 * alle drei Seiten gleichzeitig.
 *
 * complyo wird als Einzelunternehmen betrieben. Vor- und Nachname der
 * natürlichen Person sind Pflichtangabe (§ 5 Abs. 1 Nr. 1 DDG), die
 * Geschäftsbezeichnung "Complyo" allein genügt nicht. Eine Firmierung als
 * "GmbH" ohne existierende GmbH löst Rechtsscheinhaftung aus: der Handelnde
 * haftet dann persönlich, also genau umgekehrt zur Absicht einer
 * Haftungsbeschränkung.
 *
 * NICHT VERÖFFENTLICHEN, solange Pflichtfelder leer sind. Alle drei Seiten
 * weisen sichtbar darauf hin, statt still etwas Falsches zu behaupten.
 */

type Anbieterangaben = {
  /** Vor- und Nachname der natürlichen Person. Pflichtangabe. */
  name: string;
  geschaeftsbezeichnung: string;
  strasse: string;
  plz: string;
  ort: string;
  land: string;
  email: string;
  /** Postfach für Datenschutzanfragen. Muss erreichbar sein, Art. 12 Abs. 2 DSGVO. */
  datenschutzEmail: string;
  /** Zweiter Kommunikationsweg neben der E-Mail. Optional, aber üblich. */
  telefon: string;
  /** Pflichtangabe, sobald vorhanden (§ 5 Abs. 1 Nr. 6 DDG). */
  ustIdNr: string;
};

export const ANBIETER: Anbieterangaben = {
  name: '',
  geschaeftsbezeichnung: 'Complyo',
  strasse: 'Pappelallee 64',
  plz: '10437',
  ort: 'Berlin',
  land: 'Deutschland',
  email: 'info@complyo.de',
  datenschutzEmail: 'datenschutz@complyo.de',
  telefon: '',
  ustIdNr: 'DE405368946',
};

/** "10437 Berlin" */
export const ANBIETER_ANSCHRIFT = ANBIETER.plz + ' ' + ANBIETER.ort;

/** "Pappelallee 64, 10437 Berlin" — für Fließtext, etwa den Geltungsbereich der AGB. */
export const ANBIETER_ANSCHRIFT_EINZEILIG = ANBIETER.strasse + ', ' + ANBIETER_ANSCHRIFT;

/**
 * Die Vertragspartei, wie sie in den AGB steht. Ohne Namen bleibt nur die
 * Geschäftsbezeichnung; der Warnhinweis auf der Seite sagt dann, dass die
 * Angabe unvollständig ist.
 */
export const ANBIETER_VERTRAGSPARTEI = ANBIETER.name
  ? ANBIETER.name + ', handelnd unter der Geschäftsbezeichnung "' +
    ANBIETER.geschaeftsbezeichnung + '", ' + ANBIETER_ANSCHRIFT_EINZEILIG
  : ANBIETER.geschaeftsbezeichnung + ', ' + ANBIETER_ANSCHRIFT_EINZEILIG;

/** Was noch fehlt, im Klartext und mit Norm. Leer heißt vollständig. */
export const ANBIETER_FEHLENDE_PFLICHTFELDER: string[] = [
  ANBIETER.name ? null : 'Vor- und Nachname des Anbieters (§ 5 Abs. 1 Nr. 1 DDG)',
  ANBIETER.ustIdNr ? null : 'Umsatzsteuer-Identifikationsnummer (§ 5 Abs. 1 Nr. 6 DDG)',
].filter((feld): feld is string => feld !== null);

export const ANBIETER_UNVOLLSTAENDIG = ANBIETER_FEHLENDE_PFLICHTFELDER.length > 0;

File: backend/anbieter.py
# -*- coding: utf-8 -*-
"""
Anbieterdaten. Eine Quelle fuer alles, was das Backend nach aussen schreibt.

Das Gegenstueck zu `landing-react/src/lib/anbieter.ts`, das am 01.09.2026
angelegt wurde, um die erfundenen Angaben von den Rechtsseiten zu nehmen. Im
Backend blieb der alte Stand stehen — gemessen am 10.09.2026 an dreizehn
Stellen in fuenf Dateien:

  * `complyo_privacy_clause.py` schrieb "Complyo GmbH, Koburger Strasse 198,
    04416 Markkleeberg" in die Datenschutzerklaerungen DER KUNDEN. Der Kunde
    nennt damit in seiner eigenen Erklaerung einen Auftragsverarbeiter, den es
    unter diesem Namen nicht gibt (Art. 13 DSGVO).
  * Die E-Mail-Fusszeilen, die Benachrichtigungen und die PDF-Berichte nannten
    dieselbe "GmbH" — der Haftungsausschluss im Bericht lautete woertlich
    "Complyo GmbH uebernimmt keine Haftung".
  * `gdpr_api.py` nannte einer betroffenen Person als Verantwortlichen die
    "Complyo GmbH" und einen Datenschutzbeauftragten unter dpo@complyo.de.

complyo wird als Einzelunternehmen von Yvonne Weishar betrieben. Es gibt keine
GmbH. Wer ohne existierende GmbH als "GmbH" auftritt, loest Rechtsscheinhaftung
aus: der Handelnde haftet dann persoenlich — also genau umgekehrt zur Absicht
einer Haftungsbeschraenkung. Und ein Haftungsausschluss im Namen einer
Gesellschaft, die es nicht gibt, schuetzt niemanden.

Wer die Anbieterdaten aendert, aendert sie hier. `tests/test_anbieterdaten.py`
haelt fest, dass keine erfundene Rechtsform zurueckkommt.
"""

from __future__ import annotations

# Pflichtangaben nach § 5 DDG. Deckungsgleich mit dem Impressum.
NAME = "Yvonne Weishar"
GESCHAEFTSBEZEICHNUNG = "Complyo"
STRASSE = "Pappelallee 64"
PLZ = "10437"
ORT = "Berlin"
LAND = "Deutschland"
EMAIL = "info@complyo.de"
SUPPORT_EMAIL = "support@complyo.de"
DATENSCHUTZ_EMAIL = "datenschutz@complyo.de"
TELEFON = "+49 173 8448941"
UST_ID = "DE405368946"

#: "Pappelallee 64, 10437 Berlin"
ANSCHRIFT_EINZEILIG = f"{STRASSE}, {PLZ} {ORT}"

#: "Yvonne Weishar · Complyo" — kurze Nennung, etwa in einer E-Mail-Fusszeile.
ABSENDER = f"{NAME} · {GESCHAEFTSBEZEICHNUNG}"

#: Vollstaendige Nennung mit Anschrift. Fuer Fusszeilen und Berichte.
ABSENDER_MIT_ANSCHRIFT = f"{ABSENDER}, {ANSCHRIFT_EINZEILIG}"

#: Die Vertragspartei, wie sie in Vertragstexten steht.
VERTRAGSPARTEI = (
    f'{NAME}, handelnd unter der Geschäftsbezeichnung "{GESCHAEFTSBEZEICHNUNG}", '
    f"{ANSCHRIFT_EINZEILIG}"
)

#: Der Verantwortliche im Sinne des Art. 4 Nr. 7 DSGVO.
VERANTWORTLICHER = ABSENDER_MIT_ANSCHRIFT

# Es gibt keinen bestellten Datenschutzbeauftragten. Die Bestellpflicht nach
# § 38 BDSG greift erst ab 20 staendig mit der Verarbeitung beschaeftigten
# Personen. Eine erfundene Adresse dpo@complyo.de zu nennen, an die niemand
# antwortet, verletzt Art. 12 Abs. 2 DSGVO — Anfragen muessen ankommen.
DATENSCHUTZBEAUFTRAGTER: str | None = None
